runelite-client/src/main/java/net/runelite/client/plugins/runecraft/DenseRunestoneOverlay.java

```diff
@@ -0,0 +1,123 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.runecraft;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.geom.Area;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.GameObject;
+import net.runelite.api.Point;
+import net.runelite.api.Skill;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.client.game.SkillIconManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayUtil;
+
+@Slf4j
+public class DenseRunestoneOverlay extends Overlay
+{
+	private static final int MAX_DISTANCE = 2600;
+	private static final int Z_OFFSET = 200;
+	private static final int DENSE_RUNESTONE_REGION = 6972;
+
+	private static final Color CLICKBOX_BORDER_COLOR = Color.GREEN;
+	private static final Color CLICKBOX_FILL_COLOR = new Color(
+		CLICKBOX_BORDER_COLOR.getRed(), CLICKBOX_BORDER_COLOR.getGreen(),
+		CLICKBOX_BORDER_COLOR.getBlue(), 50);
+	private static final Color CLICKBOX_BORDER_HOVER_COLOR = CLICKBOX_BORDER_COLOR.darker();
+
+	private final Client client;
+	private final RunecraftPlugin plugin;
+	private final RunecraftConfig config;
+	private final SkillIconManager skillIconManager;
+
+	@Inject
+	private DenseRunestoneOverlay(
+		Client client, RunecraftPlugin plugin, RunecraftConfig config, SkillIconManager skillIconManager)
+	{
+		this.client = client;
+		this.plugin = plugin;
+		this.config = config;
+		this.skillIconManager = skillIconManager;
+
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		setPosition(OverlayPosition.DYNAMIC);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (client.getLocalPlayer().getWorldLocation().getRegionID() != DENSE_RUNESTONE_REGION)
+		{
+			return null;
+		}
+
+		boolean northStoneMineable = plugin.isDenseRunestoneNorthMineable();
+		boolean southStoneMineable = plugin.isDenseRunestoneSouthMineable();
+		GameObject northStone = plugin.getDenseRunestoneNorth();
+		GameObject southStone = plugin.getDenseRunestoneSouth();
+
+		if (northStoneMineable && northStone != null)
+		{
+			renderStone(graphics, northStone);
+		}
+		if (southStoneMineable && southStone != null)
+		{
+			renderStone(graphics, southStone);
+		}
+
+		return null;
+	}
+
+	private void renderStone(Graphics2D graphics, GameObject gameObject)
+	{
+		LocalPoint playerLocation = client.getLocalPlayer().getLocalLocation();
+		LocalPoint gameObjectLocation = gameObject.getLocalLocation();
+		if (gameObjectLocation.distanceTo(playerLocation) < MAX_DISTANCE)
```

Comment:
I'm not sure what started all of these MAX_DISTANCE things, but I'm not sure they are really required? What does it do without it?

Reply:
It would render the clickbox/indicator in the black part of the screen when you can't actually see the object yet